apps/omg/lib/omg/state.ex

```diff
@@ -172,53 +170,47 @@ defmodule OMG.State do
   end
 
   @doc """
-  Works exactly like handle_cast(:form_block) but is synchronous
-
-  Also, eth_height given is the Ethereum chain height where the block being closed got submitted, to be used with events.
+  Works exactly like handle_cast(:form_block) but:
+   - is synchronous
+   - `eth_height` given is the Ethereum chain height where the block being closed got submitted, to be used with events.
+   - relies on the caller to handle persistence, instead of handling itself
 
   Someday, one might want to skip some of computations done (like calculating the root hash, which is scrapped)
-
-  Returns `db_updates` due and relies on the caller to do persistence
   """
   def handle_call({:close_block, eth_height}, _from, state) do
-    {:ok, {_block, event_triggers, db_updates}, new_state} = do_form_block(state, eth_height)
-
-    EventerAPI.emit_events(event_triggers)
+    {:ok, {block, event_triggers, db_updates}, new_state} = do_form_block(state, eth_height)
 
+    publish_block_to_event_bus(block, event_triggers)
     {:reply, {:ok, db_updates}, new_state}
```

Comment:
one interesting thing that genserver allows you to do is not block the reply by returning
`{:reply, reply, new_state, timeout() | :hibernate | {:continue, term()}}`
so you would do something like
`{:reply, reply, new_state, {:continue, {:publish_block_to_event_bus, block, event_triggers}}}`

do you find this useful?

Reply:
did my homework on `handle_continue`, this is good to have in the toolbox!

Here however I'm not sure on needed. A small drawback is the reduced clarity of what's going on. Also both genserver messages may be bulky (it does matter, even for a sent-to-self message, doesn't it?).

Reply:
Yep, (to-self) go to the same mailbox.